PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/LinkAzureServiceBeforeRunProvider.java

```diff
@@ -0,0 +1,125 @@
+package com.microsoft.intellij.runner;
+
+import com.intellij.execution.BeforeRunTaskProvider;
+import com.intellij.execution.application.ApplicationConfiguration;
+import com.intellij.execution.configuration.AbstractRunConfiguration;
+import com.intellij.execution.configurations.ModuleBasedConfiguration;
+import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.execution.configurations.RunConfigurationModule;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.roots.ProjectFileIndex;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.microsoft.azure.toolkit.intellij.common.AzureArtifact;
+import com.microsoft.azure.toolkit.intellij.common.AzureArtifactManager;
+import com.microsoft.azure.toolkit.intellij.webapp.runner.webappconfig.WebAppConfiguration;
+import com.microsoft.azure.toolkit.lib.link.AzureLinkService;
+import com.microsoft.intellij.actions.SelectSubscriptionsAction;
+import com.microsoft.intellij.helpers.AzureIconLoader;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureIconSymbol;
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.Map;
+import java.util.Objects;
+
+public class LinkAzureServiceBeforeRunProvider extends BeforeRunTaskProvider<LinkAzureServiceBeforeRunTask> {
+    private static final Logger LOGGER = Logger.getInstance(SelectSubscriptionsAction.class);
+    private static final String NAME = "Link Azure Service";
+    private static final String DESCRIPTION = "Link Azure Service Task";
+    private static final String SPRING_BOOT_CONFIGURATION_REF = "com.intellij.spring.boot.run.SpringBootApplicationRunConfiguration";
+    public static final Key<LinkAzureServiceBeforeRunTask> ID = Key.create("LinkAzureServiceBeforeRunProviderId");
+    public static final Key<Boolean> LINK_AZURE_SERVICE = Key.create("LinkAzureService");
+    public static final Key<Map<String, String>> LINK_AZURE_SERVICE_ENVS = Key.create("LinkAzureServiceEnvs");
+
+    @Override
+    public Key<LinkAzureServiceBeforeRunTask> getId() {
+        return ID;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getDescription(LinkAzureServiceBeforeRunTask task) {
+        return DESCRIPTION;
+    }
+
+    @Nullable
+    @Override
+    public Icon getIcon() {
+        return AzureIconLoader.loadIcon(AzureIconSymbol.Common.AZURE);
+    }
+
+    @Nullable
+    @Override
+    public LinkAzureServiceBeforeRunTask createTask(@NotNull RunConfiguration runConfiguration) {
+        boolean alwaysEnable = false;
```

Comment:
whats it meaning of alwaysEnable? In my understanding, alwaysEnable means it is always true

Reply:
it means the link azure service task default to add into runconfiguration. Let me rename it to "enable".